store/src/java/com/zimbra/util/ExternalVolumeInfoHandler.java

```diff
@@ -114,17 +115,18 @@ public void deleteServerProperties(int volumeId) throws ServiceException, JSONEx
      */
     public void addServerProperties(VolumeInfo volInfo) throws ServiceException, JSONException {
         VolumeExternalInfo volExtInfo = volInfo.getVolumeExternalInfo();
+        VolumeExternalOpenIOInfo volExtOpenIoInfo = volInfo.getVolumeExternalOpenIOInfo();
+
         try {
-            // step 1: Create and update json object and array for new volume entry
+            // step 1: Create and update json object and array for new volume entry of S3
             JSONObject volExtInfoObj = new JSONObject();
-            volExtInfoObj.put(AdminConstants.A_VOLUME_ID, String.valueOf(volInfo.getId()));
-            volExtInfoObj.put(AdminConstants.A_VOLUME_STORAGE_TYPE, volExtInfo.getStorageType());
-            volExtInfoObj.put(AdminConstants.A_VOLUME_VOLUME_PREFIX, volExtInfo.getVolumePrefix());
-            volExtInfoObj.put(AdminConstants.A_VOLUME_USE_IN_FREQ_ACCESS, String.valueOf(volExtInfo.isUseInFrequentAccess()));
-            volExtInfoObj.put(AdminConstants.A_VOLUME_USE_INTELLIGENT_TIERING, String.valueOf(volExtInfo.isUseIntelligentTiering()));
-            volExtInfoObj.put(AdminConstants.A_VOLUME_GLB_BUCKET_CONFIG_ID, volExtInfo.getGlobalBucketConfigurationId());
-            volExtInfoObj.put(AdminConstants.A_VOLUME_USE_IN_FREQ_ACCESS_THRESHOLD, String.valueOf(volExtInfo.getUseInFrequentAccessThreshold()));
-
+            if( volInfo.getVolumeExternalInfo().getStorageType().equalsIgnoreCase("S3")) {
```

Comment:
For `S3` and `OpenIO` there must be constants defined, use that

Reply:
added